cEP-0022.md

```diff
@@ -0,0 +1,461 @@
+
+It is clear that successive runs of the `--lite-mode` will overestimate
+the weights in the `.quickfile`, so it seems advantageous to run `--lite-mode`
+several times and then run the **Brute Force** for the settings dropped by
+`--lite-mode` which are now in `.quickfile` to generate correct value of the
+weight. This combination of `--lite-mode` and **Brute Force** is done
+collectively by `--smart-mode`.
+
+Every run will have the annoying question in the end whether the user wants
+to answer
+a few questions about the project to get better results unless changed upon
+by the provided tags. *(For more details check out the tags section)*
+
+# Learning from real world projects
+
+## Need to perform this task
+
+There can be endless number of possibilities and assumptions:
+
+- Files having similar kind of names can be having different codestyles.
+- Files in a given directory can be following different codestyles.
+- Different parts of files maybe even following a different codestyle depending
+  on function name which can also be handled by placing `ignore`s at
+  appropriate places in the file dict.
+
+  Such kind of functionality can only be
+  added to the bears themselves, but instead we stop our train of thought over
+  here.
+
+## Making Quickstart recognize common patterns
+
+We save ourselves the additional work for building some feature that may
+hardly ever be used by any org, instead we stop assumming and start
+implementing and learning from our available resources.
+
+We now choose a list of orgs for which we try to test our `green-mode`
+ourselves. We check whether brute force is taking too much time or whether
+the `--lite-mode` or the `--smart-mode` is producing the correct results
+(i.e. *green* results). We
+choose these orgs in such a way that they are very famous or very well known or
+act as upstream repositories for a huge amount of orgs. This way a very large
+number of orgs may be mimicking the code style of these orgs itself and if
+our further optimizations can fix the problem of generating *green* config
+files for these organizations, we are in turn solving the problem for these
+other orgs/communities at the same time.
+
+We learn manually from these orgs, what combinations of these bear-setting
+values are they using, whether they are using it all over their project or only
+in certain scenarious, whether they have different codestyles in different
+directories or different nomenclature of file patterns require a different
+set of settings in their project. We give `coala-quickstart`, the ability
+to recognizing these scenarios for further
+runs.
+
+## Engaging newcomers to spread the word about coala
+
+We can only look at a finite number of orgs within the coding period so the
+last few weeks of the project should deal with writing docs and adding newcomer
+tasks for either opening PRs in other orgs of the *"green"* config files or
+feeding our repository with `QUICKINFO.yaml` files.
+# List of tags introduced
```

Comment:
You've mentioned this list quite a few times above, mind adding converting those mentions to hyperlinks to this section? Or just add "dicussed later" in braces wherever tags are mentioned.